pkg/destroy/azure/azure.go

```diff
@@ -8,6 +8,7 @@ import (
 	"strings"
 	"time"
 
+	azurestackdns "github.com/Azure/azure-sdk-for-go/services/dns/mgmt/2016-04-01/dns"
```

Comment:
Testing it again right now. Should make this simple if it works.

Reply:
Nope not working. Error message
```
DEBUG dns.ZonesClient#ListByResourceGroup: Failure responding to request: StatusCode=404 -- Original Error: autorest/azure: Service returned an error. Status=404 Code="InvalidResourceType" Message="The resource type 'dnsZones' could not be found in the namespace 'Microsoft.Network' for api version '2018-03-01-preview'. The supported api-versions are '2015-05-04-preview,2016-04-01'." 

```

Reply:
Nope does not work. `2018-03-01` does not have a dns service and I tested it with the `2018-05-01` and it gave me this error.
```
DEBUG dns.ZonesClient#ListByResourceGroup: Failure responding to request: StatusCode=404 
-- Original Error: autorest/azure: Service returned an error. Status=404 Code="InvalidResourceType" 
Message="The resource type 'dnsZones' could not be found in the namespace 
'Microsoft.Network' for api version '2018-05-01'. 
The supported api-versions are '2015-05-04-preview,2016-04-01'."       
```

Reply:
```suggestion
		azurestackdns "github.com/Azure/azure-sdk-for-go/profiles/2018-03-01/dns/mgmt/dns"
```
Then `go mod tidy` `go mod vendor` should work. But this might be a bit pedantic because it is ultimately just[ using `2016-04-01` under the hood](https://github.com/Azure/azure-sdk-for-go/blob/main/profiles/2018-03-01/dns/mgmt/dns/models.go#L14), which is what you're already using.